app/backend/src/core/controllers/appraiser/handlers/BlockRewardsHandler.ts

```diff
@@ -86,4 +88,49 @@ export class BlockRewardsHandler {
       blockRewardUsd: tipsUsd,
     };
   }
+
+  calculateBlockRewardByChainId(
+    gasUsed: bigint,
+    gasPrice: bigint,
+    baseFeePerGas: bigint,
+    priorityFeePerGas: bigint,
+    maxFeePerGas: bigint,
+  ): { gasFees: bigint; tips: bigint } {
+    const effectiveGasPrice =
+      baseFeePerGas + priorityFeePerGas != BigInt(0)
+        ? baseFeePerGas + priorityFeePerGas
+        : gasPrice;
+
+    let gasFees: bigint;
+    let tips: bigint;
+
+    switch (this.chainId) {
```

Comment:
This switch branches may better be implemented each one in a separate class (similar to `BaseHandler` inheritors), but not sure they fit into any of the existing ones?